packages/components/src/feature_alert/tooltip.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { usePopper } from 'react-popper';
+import { invoke } from 'nebenan-helpers/lib/utils';
+import CrossIcon from '@goodhood/icons/lib/16x16/cross_filled';
+
+
+import {
+  useEscHandler,
+  useOutsideClick,
+  useDelayedOpen,
+  useCloseElemClick,
+  usePopperOptions,
+} from './hooks';
+import { getTriggerProps } from './utils';
+import {
+  POSITION_TOP,
+  POSITION_BOTTOM,
+  POSITION_LEFT,
+  POSITION_RIGHT,
+  TRIGGER_HOVER,
+  TRIGGER_CLICK,
+  TRIGGER_DELAYED,
+} from './constants';
+import styles from './index.module.scss';
+
+
+const FeatureAlertTooltip = (props) => {
+  const {
+    position,
+    fallbackPosition,
+    trigger,
+    content,
+    children,
+    closeIcon,
+    defaultOpen,
+    onOpen,
+    onClose,
+    ...cleanProps
+  } = props;
+
+  const [isOpen, setOpen] = useState(defaultOpen);
+  const rootRef = useRef(null);
+  const closeIconRef = useRef(null);
+
+  const [refElement, setRefElement] = useState(null);
+  const [tooltipElement, setTooltipElement] = useState(null);
+  const [arrowElement, setArrowElement] = useState(null);
+
+  const popperOptions = usePopperOptions(arrowElement, position, fallbackPosition);
+
+  const { styles: popperStyles, attributes, forceUpdate } = usePopper(
+    refElement,
+    tooltipElement,
+    popperOptions,
+  );
+
+  // Workaround: Fix tooltip positioning when custom sizes are applied in css
+  // Positioning only breaks if container is rendered in the same cycle as
+  // `usePopper` was called for the first time.
+  useEffect(() => {
+    invoke(forceUpdate);
+  }, [forceUpdate]);
+
+  // need to be able to only open once
+  const wasActiveOnce = useRef(false);
+
+  const handleOpen = useCallback((event) => {
+    if (event) event.stopPropagation();
+    if (isOpen || wasActiveOnce.current) return;
+    wasActiveOnce.current = true;
+    setOpen(true);
+    invoke(onOpen);
+  }, [isOpen]);
+
+  const handleClose = useCallback(() => {
+    if (!isOpen) return;
+    setOpen(false);
+    invoke(onClose);
+  }, [isOpen]);
+
+  useEscHandler(handleClose);
+  useOutsideClick(rootRef, handleClose, closeIcon);
+  useCloseElemClick(closeIconRef, handleClose);
```

Comment:
One thing I don't really understand is what logic you are trying to achieve here. Can't we just put onClick on the tooltip container and hide onClick?

Reply:
Actually yes, seems like a bit more elegant, but seems like ended up being unnecessary sophisticated :)